lte/gateway/c/core/oai/tasks/amf/amf_cn.cpp

```diff
@@ -72,6 +72,32 @@ static int amf_cn_authentication_res(amf_cn_auth_res_t* const msg) {
   OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
 }
 
+//------------------------------------------------------------------------------
+static int amf_cn_implicit_deregister_ue(const amf_ue_ngap_id_t ue_id) {
+  int rc = RETURNok;
+  struct amf_context_s* amf_ctx_p = NULL;
+
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+
+  OAILOG_DEBUG(LOG_NAS_AMF,
+               "AMF-PROC Implicit Detach UE" AMF_UE_NGAP_ID_FMT "\n", ue_id);
+
+  amf_deregistration_request_ies_t params = {};
+  params.de_reg_type = AMF_NORMAL_DEREGISTRATION;
+  params.de_reg_access_type = AMF_3GPP_ACCESS;
+  params.ksi = 0;
+
+  amf_ctx_p = amf_context_get(ue_id);
+
+  if (amf_ctx_p) {
+    rc = amf_proc_deregistration_request(ue_id, &params);
+  } else {
+    OAILOG_ERROR(LOG_NAS_AMF, "Error: amf_ctx does not exits");
```

Comment:
Can you include ue_ngap_id while logging

Reply:
General comment: While logging if amf context have valid imsi64 then use OAILOG_ERROR_UE with appropriate log level